lib/resque/scheduler/scheduling_extensions.rb

```diff
@@ -149,6 +174,80 @@ def remove_schedule(name)
 
       private
 
+      def supports_lua?
+        redis_master_version >= 2.5
+      end
+
+      def redis_master_version
+        Resque.redis.info['redis_version'].to_f
+      end
+
+      def startup_sha(refresh = false)
+        @startup_sha = nil if refresh
+
+        @startup_sha ||=
+          Resque.redis.script(:load, <<-EOF.gsub(/^ {14}/, ''))
+            -- KEYS
+            -- 1: schedules
+            -- 2: persisted_schedules
+            -- 3: schedules_changed
+            -------------------------
+            -- ARGV
+            -- i:   schedule_name
+            -- i+1: encoded_schedule_config
+            -- i+2: is_schedule_persisted?
+
+            local remove_schedule = function(schedule)
+              redis.call("HDEL", KEYS[1], schedule)
```

Comment:
For readability you could do 
```lua
local schedules_key = KEYS[1]
local persisted_schedules_key = KEYS[2]
local changed_schedules_key = KEYS[3]
```
on top of the script and use those in place of the `KEYS[n]`